### DISCUSSION TITLE
How Can I get all data where nested object dosent exist?

### DISCUSSION THREAD

**Discussion by alan345** [prisma/prisma#3461] [Q&A] [answered]
Let assume that 1 product has 0 or many invoices.

How can ask Prisma to return all invoices with zero transactions?

something like? 
```ts
    const data = ctx.prisma.invoice.findMany({
      where: {
        invoices: { none: {id: ''}}
      }
    })
```

I am not very confortable with this..

**Comment by nickreynke** [comment]
Hey, so you want to get products that do not have any related invoices? I am trying to achieve the same thing here but without any success yet.

**Reply by nickreynke** [reply]
Two quick-fix solutions would be to:

1. Add an additional field `hasInvoices` and set it to `true` whenever there are invoices available.
1. Query for everything and filter with JS/TS afterwards, but this could possibly lead to unforeseen problems when using something like pagination. (Imagine you `take` 2 and those two HAVE invoices but the next two maybe wouldn't. You may end up with no results.)

I currently don't see a better solution than the first I proposed.

**Reply by ryands17** [reply]
@alan345 👋 
Could you share your schema if possible?

**Comment by alan345** [comment, marked as answer]
thanks @nickreynke,

1/ Im creating or deleting `invoice` in many places in my codebase. It means I have to also update `hasInvoices` in product which is hard to maintain.
2/I guess it is better if we do the filter straight in the database..
 


@ryands17  here you have:

```prisma
model Product {
  id                          String                               @default(cuid())
  invoices                    Invoice[]
}



model Invoice {
  id                       String                  @default(cuid()) @id
  productId                String?
  product                  Product?                @relation(fields: [productId], references: [id])
}

```



Thanks!

**Reply by alan345** [reply]
this works:
But not very elegant..

```ts
    const data = ctx.prisma.invoice.findMany({
      where: {
                NOT: [
                  {
                    product: {
                      id: {
                        not: {
                          equals: '',
                        },
                      },
                    },
                  },
                ],
      }
    })
```

**Reply by ryands17** [reply]
For products that have no invoice, you can use the following:

```ts
const data = await prisma.product.findMany({
    where: {
      invoices: {
        none: {
          id: undefined,
        },
      },
    },
})
```

And for Invoices that do not have a product associated:

```ts
const data = await prisma.invoice.findMany({
    where: {
      productId: null,
    },
})
```

**Reply by alan345** [reply]
Thank!

**Reply by alan345** [reply]
for the record, 

Get all posts with comments only? (1 post has 0, 1 or many comments)
```
    const posts = await ctx.prisma.posts.findMany({
      where: {
        NOT: {
          comments: {
            none: {
              id: undefined,
            },
          },
        },
      },
    })
```

**Reply by alan345** [reply]
or Get all posts with comments only? (1 post has 0, 1 or many comments)

```
await prisma.posts.findMany({
  where: { comments: { some: {} } },
})
```

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
